src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/FilterVariantTranches.java

```diff
@@ -52,139 +43,150 @@
+                            v.getContig(), v.getStart(), v.getEnd(),
+                            v.getReference(), v.getAlternateAlleles()));
+                }
+            }
+        }
+    }
 
     @Override
-    protected void onStartup() {
-        /* check for successful import of libraries */
-        PythonScriptExecutor.checkPythonEnvironmentForPackage("vqsr_cnn");
-        trancheString = " --tranches " + tranches.stream().map(d -> Double.toString(d)).collect(Collectors.joining(" "));
-        try {
-            final String idxExt = ".tbi";
-            tempFile = File.createTempFile(outputVcf, "_temp.vcf.gz");
-            tempFile.deleteOnExit();
-            tempFileIdx = new File(tempFile.getAbsolutePath()+idxExt);
-            tempFileIdx.deleteOnExit();
-        } catch (IOException e) {
-            throw new GATKException("Error when creating temp files.", e);
+    public void afterFirstPass() {
+        Collections.sort(snpScores, Collections.reverseOrder());
+        Collections.sort(indelScores, Collections.reverseOrder());
+
+        for(double t : tranches) {
+            int snpIndex = (int)((t/100.0)*(double)(snpScores.size()-1));
+            snpCutoffs.add(snpScores.get(snpIndex));
+            int indelIndex = (int)((t/100.0)*(double)(indelScores.size()-1));
+            indelCutoffs.add(indelScores.get(indelIndex));
         }
     }
 
     @Override
-    protected Object doWork() {
-        final Resource pythonScriptResource = new Resource("tranches.py", FilterVariantTranches.class);
-        final List<String> snpArguments = new ArrayList<>(Arrays.asList(
-                "--mode", "write_snp_tranches",
-                "--input_vcf", inputVcf,
-                "--train_vcf", snpTruthVcf,
-                "--score_keys", infoKey,
-                trancheString,
-                "--samples", Integer.toString(maxSites),
-                "--output_vcf", tempFile.getAbsolutePath()));
-
-        logger.info("SNP Args are:"+ Arrays.toString(snpArguments.toArray()));
-        final boolean pythonReturnCode = pythonExecutor.executeScript(
-                pythonScriptResource,
-                null,
-                snpArguments
-        );
-
-        final FeatureCodec<? extends Feature, ?> codec = FeatureManager.getCodecForFile(tempFile);
-        final Index index = createAppropriateIndexInMemory(codec, tempFile, tempFileIdx);
-
-        try {
-            index.write(tempFileIdx);
-        } catch (final IOException e) {
-            throw new GATKException(String.format("Could not write temporary index to file: %s", tempFileIdx.getAbsolutePath()), e);
+    protected void secondPassApply(VariantContext variant, ReadsContext readsContext, ReferenceContext referenceContext, FeatureContext featureContext) {
+        final VariantContextBuilder builder = new VariantContextBuilder(variant);
+        final double score = Double.parseDouble((String)variant.getAttribute(infoKey));
```

Comment:
`getAttribute` here and elsewhere will return null and this throw a null pointer exception when called on a variant that doesn't have the attribute.

Reply:
added check.